sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/ResourceManagerUtils.java

```diff
@@ -186,6 +186,8 @@ static String getDefaultScopeFromUrl(String url, AzureEnvironment environment) {
                     resource = String.format("https://%s/", endpoint.getValue().replaceAll("^\\.*", ""));
                     resource = removeTrailingSlash(resource);
                     break;
+                } else if (endpoint.getKey().equals(AzureEnvironment.Endpoint.STORAGE.identifier())) {
+                    resource = "https://storage.azure.com";
```

Comment:
https://github.com/Azure/azure-sdk-for-java/blob/f10cd0f9f1d9e45649bff02247c5afd774ff5f19/sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/Constants.java#L84

Reply:
Is it a default that could be overridden by user, or is it a constant that cannot be changed?

Reply:
Copied the logic from storage's PipelineBuilder: https://github.com/Azure/azure-sdk-for-java/blob/f10cd0f9f1d9e45649bff02247c5afd774ff5f19/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/util/BuilderHelper.java#L121-L126

Seems it can either be default, or the fqdn of the blob. Tested both and they all work. 
https://github.com/Azure/azure-sdk-for-java/blob/f10cd0f9f1d9e45649bff02247c5afd774ff5f19/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/BlobAudience.java#L16-L20

https://github.com/Azure/azure-sdk-for-java/blob/f10cd0f9f1d9e45649bff02247c5afd774ff5f19/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/BlobAudience.java#L33-L42

Guess the default one is good?

Reply:
Yeah, maybe fine.
Major concern is that different Azure cloud could have different scope. But if this is from storage blob, guess it's fine.

Reply:
Got it. It's a valid concern. Looked up the docs, seems all clouds share the same default `https://storage.azure.com`.

The major difference is that for scope using fqdn, it's for specific storage account and service
> Use this method to acquire a token for authorizing requests to that specific Azure Storage account and service only

Guess it's more appropriate.